### ISSUE TITLE
[BUG] Locator.or() doesn't work with iframe selectors

### ISSUE THREAD

**Issue by Nav-2d** [microsoft/playwright#23697]
### System info
- Playwright Version: 1.34.3
- Operating System: MacOS
- Browser: Chromium
- Other info:

### Source code

- [ ] I provided exact source code that allows reproducing the issue locally.

**Link to the GitHub repository with the repro**

[https://github.com/your_profile/playwright_issue_title]

or

**Config file**

```js
// playwright.config.ts
import { defineConfig, devices } from '@playwright/test';

export default defineConfig({
  projects: [
    {
      name: 'chromium',
      use: { ...devices['Desktop Chrome'], },
    },
});
```

**Test file (self-contained)**

```js
it('locator.or should work with iframe, async ({ page }) => {
    const sendButton = this.page.frameLocator('iframe').first().getByRole('button', {
      name: 'Send',
    });
    const replaceButton = this.page.frameLocator('iframe').first().getByRole('button', {
      name: 'Replace',
    })
    await expect(sendButton.or(replaceButton)).toBeVisible(); 
});
```

**Steps**
- [Run the test]

**Expected**

The tests passes with locator.or assertion

**Actual**

An error is displayed: `Error: expect.toBeVisible: Frames are not allowed inside "internal:or" selectors`

**Comment by slee628** [comment]
I see this was fixed for node.js in 1.37.1
May I ask if the same can be done for Playwright Java, please do let me know if I need to file a separate bug ticket for Java.
Thanks.

**Comment by dgozman** [comment]
@slee628 This should be already fixed across all languages in v1.36. If you still encounter the bug, please file a separate issue with a repro.
